cpp/src/arrow/filesystem/path_util.h

```diff
@@ -38,6 +39,10 @@ constexpr char kSep = '/';
 ARROW_EXPORT
 std::vector<std::string> SplitAbstractPath(const std::string& s);
 
+// Split a filename into its individual partitions.
+ARROW_EXPORT
+std::vector<std::string> SplitFilename(const std::string& s);
```

Comment:
Similar to some of the comments I made below, this method should either move to `partitioning.h` or be made more generic.  Users that are using `path_util.h` aren't going to be aware of the fact that we are splitting on `_` and even if we document it better they might be confused why such a method would exist since its purpose is very unique to partitioning.

Reply:
Made a generic `SplitAbstractPath()`